File: InPositionChatBot/InPositionChatBot.Domain/Entities/Message.cs
namespace InPositionChatBot.Domain.Entities
{
	public enum Sender
	{
		AI,
		User
	}

	public class Message : BaseEntity
	{
		public Guid Id { get; }
		public Sender Sender { get; }
		public string Text { get; }
		public DateTime Date { get; }

		public Message()
		{

		}

		public Message(Guid id, Sender sender, string text, DateTime date)
		{
			Id = id;
			Sender = sender;
			Text = text;
			Date = date;
		}
	}
}

File: InPositionChatBot.UI/src/app/chat/models/message.ts
export interface Message {
    id: string;
    sender: Sender;
    text: string;
    date: Date;
}

export interface UIMessage extends Message{
    state: State;
}

export enum State{
    success = "Success",
    error = "Error",
    noResponse = "NoResponse"
}

export enum Sender{
    AI,
    User
}